### PR TITLE
Added Kraken Functionality to the Swerve Drive Base

### PR DESCRIPTION
**Pull request by Powerlax**
Since we now have new Kraken motors that we are putting on our swerve drive base, the code must be updated to reflect this change.

This PR contains the code to change the motor type of the translation motors to from Neo motors to Kraken motors.

Most of the changes are in the KrakenSwerveModule.java file. Only the lower level methods were changed. All methods and variables that are in the SwerveModule.java and SwerveSubsystem.java and SwerveConfig.java should be in their corresponding Kraken files.

Again, I did not change the Robot Container because there are no Krakens on the swerve yet.

### CODE REVIEW

**Comment by Powerlax** [comment]
I created separate SwerveModule, SwerveSubsystem, and SwerveConfig files for the Kraken Swerve. I did this so the code was still there for normal Swerve. This is why there is a lot of duplication.

File: .vscode/settings.json
Comment (vkirub) [inline .vscode/settings.json:1]: Why are we changing vscode settings as a part of Kraken functionality changes? Ideally this should be in a separate pull request (or not included at all). 

changes are harmless, ok to proceed this time around, something to keep in mind for future.

File: google-style-formatter.xml
Comment (vkirub) [inline google-style-formatter.xml:1]: If the google-style-formatter was deleted, but it was included in the settings.json file, is that okay, are we missing something here?

File: .vscode/settings.json
Comment (Powerlax) [reply .vscode/settings.json:1]: sorry, must have slipped through from the merge conflicts

File: src/main/java/frc/robot/KrakenSwerveConfig.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import frc.robot.constants.SwerveConstants;
+
+/** Add your docs here. */
```
Comment (vkirub) [inline src/main/java/frc/robot/KrakenSwerveConfig.java:12]: Please describe what does this class do, high level summary is good.

File: src/main/java/frc/robot/KrakenSwerveConfig.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        public static final PIDController BACK_LEFT_ROT_PID = new PIDController(0.35, 0, 0);
+
+
+        public static final SwerveDriveKinematics SWERVE_KINEMATICS = new SwerveDriveKinematics(
+                        FRONT_LEFT_COORDS_METERS, FRONT_RIGHT_COORDS_METERS,
+                        BACK_LEFT_COORDS_METERS, BACK_RIGHT_COORDS_METERS);
+
+
+        // Swerve Modules and Other Hardware
+        public static final SwerveModule FRONT_LEFT_MODULE = new SwerveModule(FRONT_LEFT_ROT_ID,
```
Comment (vkirub) [inline src/main/java/frc/robot/KrakenSwerveConfig.java:80]: do these need to be public? arent you going to access them via the SWERVE_MODULES array?

File: src/main/java/frc/robot/KrakenSwerveModule.java [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        mRotMotor = new CANSparkMax(mRotID, MotorType.kBrushless);
+        mTransMotor = new TalonFX(mTransID);
+        
+        mRotEncoder = new CANcoder(rotEncoderID, "SWERVE_ENC");
+        mRotRelativeEncoder = mRotMotor.getEncoder();
+        mRotRelativeEncoder.setPosition(0);
+        mTransMotor.setPosition(0);
+
+        mRotMotor.setInverted(rotInverse);
+        mTransMotor.setInverted(transInverse);
```
Comment (vkirub) [inline src/main/java/frc/robot/KrakenSwerveModule.java:65]: May not matter much, but you may want to setInverted before you setPosition - one is configuration, and the other one is command

General rule is configure first, command later.

File: src/main/java/frc/robot/KrakenSwerveModule.java [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    /**
+     * Sets the motor speeds passed into constructor
+     * 
+     * @param desiredState takes in SwerveModule state
+     * @see SwerveModuleState
+     */
+    public void setDesiredState(SwerveModuleState desiredState) {
+
+        // Stops returning to original rotation
+        if (Math.abs(desiredState.speedMetersPerSecond) < 0.001) {
```
Comment (vkirub) [inline src/main/java/frc/robot/KrakenSwerveModule.java:172]: 0.001 is your error constant right, pull this up as a CONST easier to update globally if the system is noisy.

File: src/main/java/frc/robot/constants/KrakenSwerveConstants.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
+package frc.robot.constants;
+
+public class KrakenSwerveConstants {
+  private KrakenSwerveConstants() {
+    throw new IllegalStateException("Constants Class");
+  }
+
+  // Gear Conversions
+  public static final double TRANS_GEAR_RATIO_ROT = 1 / 6.75;
```
Comment (vkirub) [inline src/main/java/frc/robot/constants/KrakenSwerveConstants.java:9]: you are lucky that the ratio is not int 1/6 will be 0, 1 /6.75 is double. replace with 1d / 6.75

File: src/main/java/frc/robot/subsystems/KrakenSwerveSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+  }
+
+  /**
+   * 
+   * @return an array of SwerveModulePosition objects as [frontleft, frontright, backleft,
+   *         backright]
+   * @see SwerveModulePosition
+   */
+  public SwerveModulePosition[] getModulePositions() {
+    SwerveModulePosition[] positions = new SwerveModulePosition[mModules.length];
```
Comment (vkirub) [inline src/main/java/frc/robot/subsystems/KrakenSwerveSubsystem.java:129]: really hope this was a class as opposed to an array.

File: google-style-formatter.xml
Comment (geeknoid) [reply google-style-formatter.xml:1]: This needs to be undeleted.

**Comment by AadityaAgni** [comment]
In the interest of time, we will be integrating a swerve config with Kraken functionality in a future PR coming during this meeting. This system, while borrowing much of this code, is much more modular and has been tested.

**Pull request closed without merging**